nmoscommon/auth/nmos_auth.py

```diff
@@ -76,14 +78,15 @@ def get_auth_server_metadata(auth_href):
     return None
 
 
-class RequiresAuth(object):
+class AuthMiddleware(object):
 
     refresh_key_interval = 3600  # Time in Seconds until Public_key is refreshed
     public_key = None  # Shared Class Variable to share public key between instances
     key_last_refreshed = 0  # UTC time the public key was last fetched
 
-    def __init__(self, condition=OAUTH_MODE, api_name=""):
-        self.condition = condition
+    def __init__(self, app, auth_mode=OAUTH_MODE, api_name=""):
```

Comment:
As above, if the signature is changing we'll need to either check how it's being called already, or use a major bump.